### PR TITLE
Readme updates for repo & plugin

### PR DESCRIPTION
**Pull request by jtheory**
Added cloud install steps
Changed experimental to public preview

Feel free to push other changes!

### CODE REVIEW

File: src/README.md [hunk 1]
```diff
@@ -11,29 +11,26 @@ It is responsible for:
...
 ## For plugin developers
 
-This plugin is not designed to be directly interacted with; instead, use the
-convenience functions in the
-[`@grafana/experimental`](https://www.npmjs.com/package/@grafana/experimental)
+This plugin is not designed to be directly interacted with; instead, use the convenience functions
+in the [`@grafana/experimental`](https://www.npmjs.com/package/@grafana/experimental)
 package which will communicate with this plugin, if installed.
 
 First, add the correct version of `@grafana/experimental` to your dependencies in package.json:
```
Comment (yoziru) [inline src/README.md:36]: Think we should make this more generic. Minimal support is added in 1.7.0, but ideally plugin developers should aim to use the latest version >=1.7.0

File: src/README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
-# Grafana LLM app (experimental)
+# Grafana LLM app (public preview)
 
-This is a Grafana application plugin which centralizes access to LLMs across Grafana.
+This Grafana application plugin centralizes access to LLMs across Grafana.
 
 It is responsible for:
```
Comment (yoziru) [inline src/README.md:5]: Think we need to add vector search features in this section still

File: README.md [hunk 3]
```diff
@@ -1,13 +1,27 @@
...
+1. Click "Install via grafana.com"
+1. You should see your instance listed; click "Install plugin"
+1. Return to Grafana, and search installed plugins, reloading until the LLM plugin is listed (this may take a minute or two)
+1. Configuration: choose your provider (OpenAI or Azure) and fill in the fields needed
+1. Save settings, then click "Enable" (upper right) to enable the plugin
+
+
+## Install the plugin directly
 
 To install this plugin, use the `GF_INSTALL_PLUGINS` environment variable when running Grafana:
```
Comment (yoziru) [inline README.md:26]: Re this: I think the provision part is a bit out of date. We should add details on how to enable Azure OpenAI + the embedding + qdrant vectordb side. Not sure if we point to env variables or use the provisioning yaml file instead @sd2k ? 🤔

File: README.md [hunk 4]
```diff
@@ -1,13 +1,27 @@
...
-## Installing this plugin
+## Install the plugin on Grafana Cloud
+
+Prerequisites:
+- Any Grafana Cloud environment (including Free)
+- API connection details from an account with [OpenAI](https://platform.openai.com) or [Azure OpenAI](https://oai.azure.com/)
+
+Steps:
+1. In your Grafana instance, open Administration → Plugins
+1. Select "All" instead of "Installed" and search for "LLM"
```
Comment (yoziru) [inline README.md:16]: All steps have number 1 here, should we fix this numbered list?

File: README.md [hunk 5]
```diff
@@ -1,13 +1,27 @@
...
 
-## Installing this plugin
+## Install the plugin on Grafana Cloud
+
+Prerequisites:
+- Any Grafana Cloud environment (including Free)
+- API connection details from an account with [OpenAI](https://platform.openai.com) or [Azure OpenAI](https://oai.azure.com/)
+
+Steps:
+1. In your Grafana instance, open Administration → Plugins
```
Comment (yoziru) [inline README.md:15]: Would pointing to the actual link here simplify this step? (e.g. sign in -> go to https://grafana.com/grafana/plugins/grafana-llm-app/ -> install in selected instance)

File: src/README.md [hunk 6]
```diff
@@ -11,29 +11,26 @@ It is responsible for:
...
+used and requests are routed appropriately.
 
 ## For plugin developers
 
-This plugin is not designed to be directly interacted with; instead, use the
-convenience functions in the
-[`@grafana/experimental`](https://www.npmjs.com/package/@grafana/experimental)
+This plugin is not designed to be directly interacted with; instead, use the convenience functions
+in the [`@grafana/experimental`](https://www.npmjs.com/package/@grafana/experimental)
 package which will communicate with this plugin, if installed.
```
Comment (yoziru) [inline src/README.md:34]: Probably worth mentioning https://github.com/grafana/grafana-llmexamples-app here as an example repo

File: README.md (hunk 4 above)
Comment (sd2k) [reply README.md:16]: This is a common pattern - Markdown renders them all correctly either way, and it's easier to just maintain a list of `1.`s than constantly changing them as you add/remove things from the list 😂

File: README.md (hunk 3 above)
Comment (sd2k) [reply README.md:26]: `GF_INSTALL_PLUGINS` is still valid for people running Grafana in Docker but yeah we should document the provisioning yaml a lot better...

File: README.md (hunk 5 above)
Comment (sd2k) [reply README.md:15]: Yeah I think we should definitely link to that URL, saves the user a few clicks.

File: src/README.md [hunk 7]
```diff
@@ -11,29 +11,26 @@ It is responsible for:
...
-versions, or deprecated completely in favor of a different approach based on
-user feedback.
+Note: The Grafana LLM App plugin is currently in [Public preview](https://grafana.com/docs/release-life-cycle/). Grafana Labs offers support on a best-effort basis, and there might be breaking changes before the feature is generally available.
 
 ## For users
 
 Install and configure this plugin to enable various LLM-related functionality across Grafana.
-This will include new functionality inside Grafana itself, such as explaining panels, or
-in plugins, such as natural language query editors.
+This include new functionality inside Grafana itself, such as explaining panels, or in plugins,
```
Comment (sd2k) [inline src/README.md:24]:
```suggestion
This includes new functionality inside Grafana itself, such as explaining panels, or in plugins,
```

File: src/README.md (hunk 1 above)
Comment (sd2k) [reply src/README.md:36]: Yeah I think we should just say 'add the latest version of `@grafana/experimental`', no reason for anyone to use a specific version any more.

File: src/README.md (hunk 2 above)
Comment (jtheory) [reply src/README.md:5]: We will -- I left it out just while all those pieces are landing, and figured we'll update the docs again when ready.

File: README.md [hunk 8]
```diff
@@ -1,13 +1,27 @@
-# Grafana LLM App (Experimental)
+# Grafana LLM App (Public Preview)
```
[the remark this answers is not in the document]
Comment (jtheory) [reply README.md:1]: Good question -- I created a walk-through video setting up this plugin for #dashboard-ai to help enable Sales & CX but that's all, and I'm not sure exactly what the Grafana FE squad has done to support enablement.

File: README.md (hunk 3 above)
Comment (jtheory) [reply README.md:26]: Ben merged in updates - recheck?

File: src/README.md [hunk 9]
```diff
@@ -11,29 +11,26 @@ It is responsible for:
...
 ## For plugin developers
 
-This plugin is not designed to be directly interacted with; instead, use the
-convenience functions in the
-[`@grafana/experimental`](https://www.npmjs.com/package/@grafana/experimental)
+This plugin is not designed to be directly interacted with; instead, use the convenience functions
+in the [`@grafana/experimental`](https://www.npmjs.com/package/@grafana/experimental)
 package which will communicate with this plugin, if installed.
 
 First, add the correct version of `@grafana/experimental` to your dependencies in package.json:
```
Comment (jtheory) [inline src/README.md:36]:
```suggestion
First, add the latest version of `@grafana/experimental` to your dependencies in package.json:
```

File: src/README.md (hunk 2 above)
Comment (jtheory) [reply src/README.md:5]: Discussion in Slack about docs & separating these 2 releases / iterations

File: src/README.md [hunk 10]
```diff
@@ -11,32 +11,29 @@ It is responsible for:
...
+used and requests are routed appropriately.
 
 ## For plugin developers
 
-This plugin is not designed to be directly interacted with; instead, use the
-convenience functions in the
-[`@grafana/experimental`](https://www.npmjs.com/package/@grafana/experimental)
+This plugin is not designed to be directly interacted with; instead, use the convenience functions
+in the [`@grafana/experimental`](https://www.npmjs.com/package/@grafana/experimental)
 package which will communicate with this plugin, if installed.
```
Comment (jtheory) [inline src/README.md:34]:
```suggestion
package which will communicate with this plugin, if installed.

Looking for working examples? Check https://github.com/grafana/grafana-llmexamples-app
```

File: README.md (hunk 5 above)
Comment (jtheory) [reply README.md:15]: It is a little faster, but they're going to need to find the LLM plugin inside their instance anyway to confirm the install worked, and then to configure it. :shrug:

I am wondering if it's not great for us to walk through plugin installation steps when they are elsewhere in docs (and might change in the future and break our steps here): https://grafana.com/docs/grafana/latest/administration/plugin-management/#install-a-plugin

File: README.md [hunk 11]
```diff
@@ -46,6 +60,77 @@ apps:
...
+
+where:
+
+- `<resource>` is your Azure OpenAI resource name
+- the `azureModelMapping` field contains `[model, deployment]` pairs so that features know
+  which Azure deployment to use in place of each model you wish to be used.
+
+### Provisioning vector services
+
+The vector services of the plugin allow certain AI-based features to perform semantic search to improve the responses given by LLMs. Configuration is in roughly three parts:
```
Comment (jtheory) [inline README.md:92]: We may want to skip the vector service in docs until the pieces are in place

File: README.md [hunk 12]
```diff
@@ -1,13 +1,27 @@
...
+
+Prerequisites:
+- Any Grafana Cloud environment (including Free)
+- API connection details from an account with [OpenAI](https://platform.openai.com) or [Azure OpenAI](https://oai.azure.com/)
+
+Steps:
+1. In your Grafana instance, open Administration → Plugins
+1. Select "All" instead of "Installed" and search for "LLM"
+1. Click "Install via grafana.com"
+1. You should see your instance listed; click "Install plugin"
```
Comment (jtheory) [inline README.md:18]:
```suggestion
1. On the [LLM's plugin page](https://grafana.com/grafana/plugins/grafana-llm-app/), you should see your instance listed; click "Install plugin"
```

File: README.md (hunk 5 above)
Comment (jtheory) [reply README.md:15]: I think I found a compromise -- I included the direct plugin link in the steps, so someone who knows how this all works can skip ahead. Thoughts?

Comment (sd2k) [reply README.md:15]: Looks good to me, I think I'd be able to follow these instructions 😅

File: README.md (hunk 11 above)
Comment (sd2k) [reply README.md:92]: TBH I think the pieces _are_ in place, but if you'd like we can do two separate releases? Feel free to remove this section if so

**Review by sd2k — APPROVED** [review]

File: README.md [hunk 13]
```diff
@@ -46,6 +60,33 @@ apps:
...
+where:
+
+- `<resource>` is your Azure OpenAI resource name
+- the `azureModelMapping` field contains `[model, deployment]` pairs so that features know
+  which Azure deployment to use in place of each model you wish to be used.
+
+
 ## Adding LLM features to your plugin or Grafana core
 
 To make use of this plugin when adding LLM-based features, you can use the helper functions in the `@grafana/experimental` package.
```
Comment (yoziru) [inline README.md:92]: The line below here still says "correct version", could we use "latest" version as in https://github.com/grafana/grafana-llm-app/blob/4b941a67aeb4ce438bf62f4c10a27b8e7c8be72e/src/README.md#for-plugin-developers?

**Review by yoziru — APPROVED** [review]
One tiny nit, otherwise g2g for me

File: README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (csmarchbanks) [reply README.md:1]: We decided to stick with Public Preview, resolving this.

**Review by edwardcqian — APPROVED** [review]
lgtm

File: README.md (hunk 13 above)
Comment (jtheory) [reply README.md:92]: may just need to refresh the page: 
image

**Pull request merged**